lib/m2ee/munin.py

```diff
@@ -104,35 +105,63 @@ def print_config(m2ee, name):
 
 
 def print_values(m2ee, name):
-    stats, java_version = get_stats('values', m2ee.client, m2ee.config)
+    stats, java_version = get_stats("values", m2ee.client, m2ee.config)
     if stats is None:
         return
     stats = augment_and_fix_stats(stats, m2ee.runner.get_pid(), java_version)
     options = m2ee.config.get_munin_options()
 
     print_requests_values(name, stats)
     print_connectionbus_values(name, stats)
-    print_sessions_values(name, stats, options.get('graph_total_named_users', True))
+    print_sessions_values(
+        name, stats, options.get("graph_total_named_users", True)
+    )
     print_jvmheap_values(name, stats)
     print_threadpool_values(name, stats)
     print_cache_values(name, stats)
     print_jvm_threads_values(name, stats)
-    print_jvm_process_memory_values(name, stats, m2ee.runner.get_pid(), m2ee.client, java_version)
+    print_jvm_process_memory_values(
+        name, stats, m2ee.runner.get_pid(), m2ee.client, java_version
+    )
 
 
 def guess_java_version(client, runtime_version, stats):
-    m2eeresponse = client.about()
-    if not m2eeresponse.has_error():
-        about = m2eeresponse.get_feedback()
-        if 'java_version' in about:
-            java_version = about['java_version']
-            java_major, java_minor, _ = java_version.split('.')
-            return int(java_minor)
+    m2ee_response = client.about()
+    return _guess_java_version(m2ee_response, runtime_version, stats)
+
+
+def _get_jre_major_version_from_version_string(version_string):
+    """Java changed their versioning scheme for JRE/JDK greater than 9,
+    as per https://openjdk.java.net/jeps/223
+    """
+    # *_ captures all remaining tuple elements
+    major, minor, *_ = version_string.split(".")
+    if major == "1":
+        return int(minor)
+    return int(major)
+
+
+def _guess_java_version(m2ee_response, runtime_version, m2ee_stats):
+    # type: ("lib.m2ee.client.M2EEResponse", "lib.m2ee.version.MXVersion", dict) -> "Optional[int]"
+    """"This internal function has a more unit-testable API than
+    `guess_java_version`, which enables us to preserve compatibility, whilst
+    simultaneously adding unit testing.
+    """
+    if not m2ee_response.has_error():
+        about = m2ee_response.get_feedback()
+        if "java_version" in about:
+            java_version_string = about["java_version"]
+            return _get_jre_major_version_from_version_string(
+                java_version_string
+            )
+
+    # These branches should never be reached for Mendix versions in the Mendix
```

Comment:
I wrote this comment, but I think that it's not actually accurate. You can run all Mendix 6 versions in Cloud V4, some of which lack the "java_version" field.

I think that this comment should only refer to line 162,  `if runtime_version // 5`

Can you verify this with an early Mendix 6 version and share the m2ee about output? (The "188v4test" application in the test environment has a Mendix 6.5 MDS deployed to the "test" environment of that application, that might be a good starting point).

Reply:
@joshp123 This is how the about output looks like : 
```
{'feedback': {'build': '8913',
              'copyright': 'Copyright © 2003-2016 Mendix bv. All rights '
                           'reserved.',
              'model_version': 'unversioned',
              'name': 'Mendix Runtime',
              'vendor': 'Mendix',
              'version': '6.5.0',
              'xasid': 'b82e1c10-dd51-4a0d-8998-e1361d903ca1'},
 'result': 0}
```

Yeah, I don't see the `java_version` string, but we would reach that branch of course. I'll move the comment to the line below.
I'll move the line for runtime_version

Reply:
I've ported the relevant commit changes from #320 which has the appropriate comments.